python/metatensor-operations/metatensor/operations/testing/_grad.py

```diff
@@ -0,0 +1,73 @@
+from typing import Callable
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+from .._classes import TensorMap
+
+
+def finite_differences(
```

Comment:
Or maybe I generalize this to use the length of the components in the gradients for this first for loop then the name is suitable and this can be used for more general tests